File: wikipedia_sample.py
def main():
  pages = {}
  links = {}

  with open('data/pages_small.txt') as f:
    for data in f.read().splitlines():
      page = data.split('\t')
      # page[0]: id, page[1]: title
      pages[page[0]] = page[1]

  with open('data/links_small.txt') as f:
    for data in f.read().splitlines():
      link = data.split('\t')
      # link[0]: id (from), links[1]: id (to)
      if link[0] in links:
        links[link[0]].add(link[1])
      else:
        links[link[0]] = {link[1]}

  for k, v in pages.items():
    if v == 'Google':
      print('Google', k)


if __name__ == '__main__':
  main()

File: WikipediaSample.java
import java.io.*;
import java.util.*;

public class WikipediaSample {
    public static void main(String[] args) {
        Map<String, String> pages = new TreeMap<String, String>();
        Map<String, Set<String>> links = new TreeMap<String, Set<String>>();

        try {
            File pageFile = new File("data/pages_small.txt");
            Scanner pageReader = new Scanner(pageFile);
            while (pageReader.hasNextLine()) {
                String[] page = pageReader.nextLine().split("\t", 0);
                // page[0]: id, page[1]: title
                pages.put(page[0], page[1]);
            }
            pageReader.close();

            File linkFile = new File("data/links_small.txt");
            Scanner linkReader = new Scanner(linkFile);
            while (linkReader.hasNextLine()) {
                String[] link = linkReader.nextLine().split("\t", 0);
                // link[0]: id (from), links[1]: id (to)
                if (!links.containsKey(link[0])) {
                    links.put(link[0], new TreeSet<String>());
                }
                links.get(link[0]).add(link[1]);
            }
            linkReader.close();
        } catch (FileNotFoundException e) {
            System.out.println("File not found.");
            return;
        }

        for (Map.Entry<String, String> page : pages.entrySet()) {
            if (page.getValue().equals("Google")) {
                System.out.println(page.getValue() + " " + page.getKey());
                break;
            }
        }
    }
}
